app/domain/authentication/authn_jwt/identity_providers/identity_from_url_provider.rb

```diff
@@ -0,0 +1,28 @@
+module Authentication
+  module AuthnJwt
+    module IdentityProviders
+      # Provides jwt identity from information in the URL
+      class IdentityFromUrlProvider < IdentityProviderInterface
+        def initialize(authentication_parameters:)
+          @authentication_parameters = authentication_parameters
+        end
+
+        def jwt_identity
+          raise Errors::Authentication::AuthnJwt::IdentityMisconfigured unless identity_available?
+
+          @authentication_parameters.username
+        end
+
+        def identity_available?
+          return @identity_available if defined?(@identity_available)
+
+          @identity_available = username_exists?
+        end
+
+        def username_exists?
```

Comment:
this method should be private. it will also help us in the (hopefully) upcoming refactor.

Reply:
Done